src/qt/anybutton.cpp

```diff
@@ -45,61 +46,67 @@ bool wxQtPushButton::eventFilter(QObject *watched, QEvent *event)
     QPushButton *button = qobject_cast<QPushButton *>( watched );
     wxAnyButton *wxbutton = GetHandler();
     QIcon icon = button->icon();
-    if( event->type() == QEvent::HoverMove && button->isEnabled() )
-    {
-        if( wxbutton->GetStateBitmaps()[1].IsOk() )
-        {
-            button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[1].GetHandle() ) ) );
-        }
-    }
-    else if( event->type() == QEvent::MouseButtonPress && button->isEnabled() )
-    {
-        if( wxbutton->GetStateBitmaps()[2].IsOk() )
-        {
-            button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[2].GetHandle() ) ) );
-        }
-    }
-    else if( event->type() == QEvent::FocusIn && button->isEnabled() )
-    {
-        if( wxbutton->GetStateBitmaps()[4].IsOk() )
-            button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[4].GetHandle() ) ) );
-    }
-    else if( event->type() == QEvent::EnabledChange )
-    {
-        if( !button->isEnabled() && wxbutton->GetStateBitmaps()[3].IsOk() )
-            button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[3].GetHandle() ) ) );
-        else
-        {
-            if( button->hasFocus() && wxbutton->GetStateBitmaps()[4].IsOk() )
-                button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[4].GetHandle() ) ) );
-            if( !button->hasFocus() && wxbutton->GetStateBitmaps()[0].IsOk() )
-                button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[0].GetHandle() ) ) );
-        }
-    }
-    else if( event->type() == QEvent::HoverLeave )
-    {
-        if( button->hasFocus() && wxbutton->GetStateBitmaps()[4].IsOk() )
-            button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[4].GetHandle() ) ) );
-        else if( !button->isEnabled() && wxbutton->GetStateBitmaps()[3].IsOk() )
-            button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[3].GetHandle() ) ) );
-        else if( wxbutton->GetStateBitmaps()[0].IsOk() )
-            button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[0].GetHandle() ) ) );
-    }
-    else if( event->type() == QEvent::FocusOut )
+
+    switch ( event->type() )
     {
-        if( !button->isEnabled() && wxbutton->GetStateBitmaps()[3].IsOk() )
-            button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[3].GetHandle() ) ) );
-        else if( wxbutton->GetStateBitmaps()[0].IsOk() )
-            button->setIcon( QIcon( *( wxbutton->GetStateBitmaps()[0].GetHandle() ) ) );
+        case QEvent::HoverMove:
+            if( button->isEnabled() )
+                checkAndSetIcon(button, wxbutton, 1);
```

Comment:
Hi,
Thank you for working on this.
I think you should get rid of the "magic numbers" here and use some define constants.